src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+const listItems = [...document.querySelectorAll('li')];
+const employeesList = document.querySelector('ul');
+
+function convertToNumber(string) {
+  const salaryNumber = string.replace(/\$/g, '').replace(/,/g, '');
+
+  return +salaryNumber;
+}
+
+function sortBySalary(items) {
+  items.sort(
+    (a, b) =>
+      convertToNumber(b.dataset.salary) - convertToNumber(a.dataset.salary),
+  );
+
+  return items;
+}
+
+function updateList(list, items) {
+  list.innerHTML = '';
+
+  const fragment = document.createDocumentFragment();
+
+  items.forEach((item) => fragment.appendChild(item));
+
+  list.appendChild(fragment);
+}
+
+sortBySalary(listItems);
+updateList(employeesList, listItems);
```

Comment:
Code Quality: The sorted array is not assigned to any variable before being passed to the updateList function. This could lead to confusion about the state of the listItems array. It may be better to assign the sorted array to a new variable and then pass that variable to the updateList function.